src/Cake.Core/CakeTaskBuilderExtensions.cs

```diff
@@ -30,6 +30,26 @@ public static CakeTaskBuilder<T> IsDependentOn<T>(this CakeTaskBuilder<T> builde
         }
 
         /// <summary>
+        /// Creates a dependency between two tasks.
+        /// </summary>
+        /// <typeparam name="T">The task type.</typeparam>
+        /// <typeparam name="TOther">The task type that this task depends on.</typeparam>
+        /// <param name="builder">The task builder.</param>
+        /// <param name="taskBuilder">The name of the dependent task.</param>
+        /// <returns>The same <see cref="CakeTaskBuilder{T}"/> instance so that multiple calls can be chained.</returns>
+        public static CakeTaskBuilder<T> IsDependentOn<T, TOther>(this CakeTaskBuilder<T> builder, CakeTaskBuilder<TOther> taskBuilder)
+            where T : CakeTask where TOther : CakeTask
+        {
+            if (builder == null)
```

Comment:
Could you add a test that verifies that the method will throw if `builder` is null.